service/src/main/java/bio/terra/externalcreds/controllers/FenceAccountKeyController.java

```diff
@@ -0,0 +1,57 @@
+package bio.terra.externalcreds.controllers;
+
+import bio.terra.externalcreds.auditLogging.AuditLogEvent;
+import bio.terra.externalcreds.auditLogging.AuditLogEventType;
+import bio.terra.externalcreds.auditLogging.AuditLogger;
+import bio.terra.externalcreds.generated.api.FenceAccountKeyApi;
+import bio.terra.externalcreds.generated.model.Provider;
+import bio.terra.externalcreds.models.LinkedAccount;
+import bio.terra.externalcreds.services.FenceAccountKeyService;
+import bio.terra.externalcreds.services.LinkedAccountService;
+import jakarta.servlet.http.HttpServletRequest;
+import java.time.Instant;
+import java.util.Optional;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+@Controller
+public record FenceAccountKeyController(
+    AuditLogger auditLogger,
+    ExternalCredsSamUserFactory samUserFactory,
+    FenceAccountKeyService fenceAccountKeyService,
+    HttpServletRequest request,
+    LinkedAccountService linkedAccountService)
+    implements FenceAccountKeyApi {
+
+  @Override
+  public ResponseEntity<String> getFenceAccountKey(Provider provider) {
+    var samUser = samUserFactory.from(request);
+    var maybeLinkedAccount =
+        linkedAccountService.getLinkedAccount(samUser.getSubjectId(), provider);
+    var maybeFenceAccountKey =
+        fenceAccountKeyService.getFenceAccountKey(samUser.getSubjectId(), provider);
+
+    auditLogger.logEvent(
```

Comment:
If the linked account doesn't exist, wouldn't we want to just return a 404? We definitely don't want to log an event if there's no linked account.